packages/plugin-ext/src/common/plugin-api-rpc.ts

```diff
@@ -1594,13 +1594,26 @@ export interface StorageExt {
     $updatePluginsWorkspaceData(data: KeysToKeysToAnyValue): void;
 }
 
+export enum DebugConfigurationProviderTriggerKind {
+    /**
+     * `DebugConfigurationProvider.provideDebugConfigurations` is called to provide the initial debug
+     * configurations for a newly created launch.json.
+     */
+    Initial = 1,
+    /**
+     * `DebugConfigurationProvider.provideDebugConfigurations` is called to provide dynamically generated
+     * debug configurations when the user asks for them through the UI (e.g. via the "Select and Start Debugging" command).
+     */
+    Dynamic = 2
+}
```

Comment:
I find these docs a little hard to interpret. Maybe add a JS for the whole `enum` explaining where this is used. Something like:

```typescript
/**
 * A value passed to <something> when <something> to determine when a `DebugConfigurationProvider`'s `provideDebugConfigurations` method should be called.
 */

(The explanation below around the register method is very clear.)

Reply:
Done!
I have aligned similar comments under `theia.d.ts` and `plugin-api-rpc.ts`